docs/messenger/messenger.rst

```diff
@@ -0,0 +1,122 @@
+- **Based on Symfony Messenger Component**: The Messenger feature is built upon the Symfony's Messenger component, offering robust performance and features.
+
+Existing Mautic customers on version 3.x or 4.x who have used the now obsolete QueueBundle would need to configure Messenger to continue using it. Also, Messenger is already implemented for email (this document does not cover it), allowing users to configure it in Mautic's UI settings.
+
+Configuration
+=============
+
+.. note::
+
+    The default configuration needs no changes and will work out of the box. With the default settings, the messages will be processed synchronously and no messenger will be used.
+
+Transports
+----------
+
+A transport is a connection to an endpoint that can receive and serve messages (understand work payload). For example, a synchronous transport has no endpoint and instead, the handler is called directly. In comparison, the RabbitMQ transport (AMQP) uses a RabbitMQ server as an endpoint and will send and fetch these messages from remote server. Doctrine transport uses a database.
+
+For more information, refer to the `Symfony documentation <https://symfony.com/doc/current/messenger.html#transports>`_.
+
+Failed Transport
+^^^^^^^^^^^^^^^^
+
+Failed transport in Symfony Messenger is a feature that helps with managing failed messages. A failed message is a message that could not be handled by a message handler due to an exception or error.
+
+Symfony Messenger automatically catches exceptions thrown by handlers. Depending on the transport configuration, the message can either be retried or sent to the "failure transport".
+
+This failure transport serves as a storage for these failed messages and allows you to view, debug, and retry the processing of these messages manually.
+
+
+Routing
+-------
+
+Every message needs to have a transport associated with it. The synchronous transport is a valid selection. By default, PageHit and EmailHit messages are routed to the **hit** transport, which is configured as synchronous.
+
+.. code-block:: php
+
+    MauticMessengerTransports::HIT    => [
+        'dsn'        => 'sync://',
+        'serializer' => 'messenger.transport.jms_serializer',   // Smaller payload
+    ],
+
+.. _default-configuration:
+
+The default configuration is as follows
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: php
+
+    'routing' => [
+        \Symfony\Component\Mailer\Messenger\SendEmailMessage::class => MauticMessengerTransports::EMAIL,
+        \Mautic\MessengerBundle\Message\PageHitNotification::class  => MauticMessengerTransports::HIT,
+        \Mautic\MessengerBundle\Message\EmailHitNotification::class => MauticMessengerTransports::HIT,
+    ],
```

Comment:
What is this information for? Will users need to configure it themselves or change it?

Reply:
it is to show how routing is done so it is easier to understand why we configure just the one transport a little lower. so they understand the routing
message => route => transport => handler